src/Task.cc

```diff
@@ -1234,24 +1251,25 @@ static bool cpu_has_KNL_string_singlestep_bug() {
 static int single_step_coalesce_cutoff() { return 16; }
 
 void Task::work_around_KNL_string_singlestep_bug() {
-  uintptr_t cx = regs().cx();
-  uintptr_t cutoff = single_step_coalesce_cutoff();
   /* The extra cx >= cutoff check is just an optimization, to avoid the
      moderately expensive load from ip() if we can */
-  if (cpu_has_KNL_string_singlestep_bug() && cx > cutoff &&
-      at_x86_string_instruction(this)) {
-    /* KNL has a quirk where single-stepping a string instruction can step up
-       to 64 iterations. Work around this by fudging registers to force the
-       processor to execute one iteration and one interation only. */
-    LOG(debug) << "Working around KNL single-step hardware bug (cx=" << cx
-               << ")";
-    if (cx > cutoff) {
-      last_resume_orig_cx = cx;
-      Registers r = regs();
-      /* An arbitrary value < cutoff would work fine here, except 1, since
-         the last iteration of the loop behaves differently */
-      r.set_cx(cutoff);
-      set_regs(r);
+  if (cpu_has_KNL_string_singlestep_bug()) {
```

Comment:
I think it'd be easier just to `if (!cpu_has_KNL_string_singlestep_bug()) { return; }` at the beginning and leave the rest of this function undisturbed.